tests/lax_numpy_test.py

```diff
@@ -4796,10 +4820,17 @@ def testArangeJit(self):
     expected = jtu.with_jax_dtype_defaults(np.arange)(5)
     self.assertAllClose(ans, expected)
 
-  @jtu.sample_product(args=[(5,), (0, 5)])
-  def testArangeJaxpr(self, args):
-    jaxpr = jax.make_jaxpr(lambda: jnp.arange(*args))()
-    self.assertEqual(len(jaxpr.jaxpr.eqns), 1)
+  @jtu.sample_product(
+    args=[(5,), (0, 5)],
+    kwargs=[{"device": None}, {"device": jax.devices()[-1]}]
```

Comment:
You can't call `jax.devices()` at the top-level in tests, because in absl environments the devices won't be available until `app.run()` is called at the bottom of the script.

Can you put the `jax.devices()[-1]` call into the body of the function?

Reply:
I see, how about this : 
```python
  @jtu.sample_product(
    args=[(5,), (0, 5)],
    kwargs=[{"device": None}, {"device": "last"}]
  )
  def testArangeJaxpr(self, args, kwargs):
    if kwargs["device"] == "last":
      kwargs["device"] = jax.devices()[-1]
    jaxpr = jax.make_jaxpr(lambda: jnp.arange(*args, **kwargs))()
```
If this is too ugly, I can make a for-loop inside the test

Reply:
How about `specify_device=[True, False]` and then in the function use `device = devices()[-1] if specify_device else None`